File: app/src/main/java/com/hitstudio/apps/sharednotebook/domain/repository/NoteRepository.java
package com.hitstudio.apps.sharednotebook.domain.repository;

import androidx.lifecycle.LiveData;

import com.hitstudio.apps.sharednotebook.domain.model.Note;

import java.util.List;

public interface NoteRepository {
    LiveData<List<Note>> getAllNotes();
    Note getNoteById(String id);
    void insertNote(Note note);
    void updateNote(Note note);
    void deleteNote(Note note);
}

File: app/src/main/java/com/hitstudio/apps/sharednotebook/domain/repository/NoteRepository.kt
package com.hitstudio.apps.sharednotebook.domain.repository

import com.hitstudio.apps.sharednotebook.domain.model.Note
import kotlinx.coroutines.flow.Flow

interface NoteRepository {
    fun getAllNotes(): Flow<List<Note>>
    suspend fun getNoteById(id: String): Note?
    suspend fun insertNote(note: Note)
    suspend fun updateNote(note: Note)
    suspend fun deleteNote(note: Note)
}